smithy-aws-iam-traits/src/main/java/software/amazon/smithy/aws/iam/traits/ConditionKeyDefinition.java

```diff
@@ -91,13 +103,16 @@ public Optional<String> getRelativeDocumentation() {
         return Optional.ofNullable(relativeDocumentation);
     }
 
+    public Optional<Boolean> getRequired() { return Optional.ofNullable(required); }
```

Comment:
Since `required` is a primitive boolean, it's not nullable. This should just be a `boolean isRequired()` method that returns the value.

Reply:
Updated to be isRequired without nullable